### DISCUSSION TITLE
there is not enough being logged when having an index conflict in mongodb

### DISCUSSION THREAD

**Discussion by Loki-Afro** [mikro-orm/mikro-orm#5511] [General]
### Describe the bug

there is not enough being logged when having an index conflict in mongodb, the actual reason is hidden, like an index option conflict

![image](https://github.com/mikro-orm/mikro-orm/assets/4738445/8ed00bd9-dbc2-48d0-a6ef-08293cdc64bc)

the code in question: https://github.com/mikro-orm/mikro-orm/blob/v5.6.16/packages/mongodb/src/MongoSchemaGenerator.ts#L128

### Reproduction

create a faulty index

### What driver are you using?

@mikro-orm/mongodb

### MikroORM version

5.6.16

### Node.js version

18.20.2

### Operating system

linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> 5.6.16

That’s ancient, have you tried v6?

**Comment by Loki-Afro** [comment]
@B4nan why is it relevant if the code hasn't changed in that regard?
https://github.com/mikro-orm/mikro-orm/blob/v6.2.3/packages/mongodb/src/MongoSchemaGenerator.ts#L141

**Reply by B4nan** [reply]
Is this #4486 trying to add what you are missing?

**Reply by Loki-Afro** [reply]
looks like it would do the trick, although I'm not a typescript expert.
